### ISSUE TITLE
hub alias hides git-alias

### ISSUE THREAD

**Issue by cdlm** [github/hub#153]
`hub alias` does not seem to take `--noop` into account, so it hides `git-alias` from https://github.com/visionmedia/git-extras.

Since it's a hub-specific command, and it's only useful at setup time, I propose to either:
- make it aware of `--noop`
- hide it as an option rather than a command, i.e. `hub [--alias|--shell-alias] [SHELL]`
- make it work only if `$0` is explicitly `hub`

**Comment by abhisekp** [comment]
@cdlm @mislav Does hub make git-extras aliases useless if I've git aliased to hub? Can't I use any of the awesome git-extras aliases mentioned here https://github.com/tj/git-extras/blob/master/Commands.md ?

**Comment by mislav** [comment]
You can use all of the git-extras command except for `git alias` and `git fork` from that repo, because `hub alias` and `hub fork` will take precedence.

**Comment by mcornella** [comment]
You can use `git-alias` and `git-fork` instead, or bypass the `git` alias with `\` (backslash): `\git alias`. I'm not a go expert, but I've been messing around and no matter how I run the go executable `os.Args[0]` always gives the true name of the executable, which is `hub`, as @mislav said.

In the case of `hub alias` perhaps you can use the fact that `-s` is passed to run `hub alias`, otherwise print a deprecation notice or forward over to the `git` command.

**Comment by mislav** [comment]
For any command if you want to skip `hub` altogether from your shell, just do `command git ...`. E.g. `command git alias`.

**Comment by mcornella** [comment]
Yep, and that has the same effect as using backslash `\`, *e.g.* `\git`

**Comment by jacobj10** [comment]
@github
